internal/app/machined/pkg/controllers/kubeaccess/crd.go

```diff
@@ -0,0 +1,207 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package kubeaccess
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/go-pointer"
+	"github.com/talos-systems/crypto/x509"
+	"go.uber.org/zap"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/talos-systems/talos/internal/app/machined/pkg/controllers/kubeaccess/talossa"
+	"github.com/talos-systems/talos/internal/pkg/etcd"
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+	"github.com/talos-systems/talos/pkg/machinery/resources/config"
+	"github.com/talos-systems/talos/pkg/machinery/resources/kubeaccess"
+	"github.com/talos-systems/talos/pkg/machinery/resources/secrets"
+)
+
+// CRDController manages Kubernetes endpoints resource for Talos API endpoints.
+type CRDController struct{}
+
+// Name implements controller.Controller interface.
+func (ctrl *CRDController) Name() string {
+	return "kubeaccess.CRDController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *CRDController) Inputs() []controller.Input {
+	return nil
+}
+
+// Outputs implements controller.Controller interface.
+func (ctrl *CRDController) Outputs() []controller.Output {
+	return nil
+}
+
+// Run implements controller.Controller interface.
+//
+//nolint:gocyclo,cyclop
+func (ctrl *CRDController) Run(ctx context.Context, r controller.Runtime, logger *zap.Logger) error {
+	if err := r.UpdateInputs([]controller.Input{
```

Comment:
As these inputs are static, can we move them to `Inputs()` return?
Or add the comment why we need it in the `Run` call.

Reply:
I did it this way because @smira did it the same way in `endpoint.go` - maybe he can comment on this.